src/library/scala/concurrent/ExecutionContext.scala

```diff
@@ -139,7 +139,25 @@ object ExecutionContext {
    *
    * @return the global `ExecutionContext`
    */
-  def global: ExecutionContextExecutor = Implicits.global.asInstanceOf[ExecutionContextExecutor]
+  final def global: ExecutionContextExecutor = Implicits.global.asInstanceOf[ExecutionContextExecutor]
+
+  /**
+   * The `callingThread` `ExecutionContext` executes `Runnable`s on the calling `Thread`.
+   * Nested submissions will be trampolined to prevent uncontrolled stack space growth.
+   * Any `NonFatal` or `InterruptedException`s will be reported to the `defaultReporter`.
+   *
+   * In order to maximize chances of maintaining application responsiveness, it is strongly
+   * advised to only execute logic which will quickly return control to the caller.
+   *
+   * DO NOT call any blocking code in the `Runnable`s submitted to this `ExecutionContext`
+   * as it will prevent progress by other enqueued `Runnable`s and the calling `Thread`.
```

Comment:
I like that the potential performance improvements from this over a thread pool executor are not mentioned - the last thing we want is people seeing this and saying "oh magic performance improvement secret sauce! I'm going to use this!" without reading the rest of the docs. Also, I wonder if we can be even clearer about the need for care. Something like "Symptoms of misusing this `ExecutionContext` include deadlocks and a potential for your application to slow to a crawl under minimal load."

Reply:
@jroper Great points, James. I'll craft a message about symptoms of misuse.

Reply:
@jroper The new working name is `parasitic`—please check out the new docs. :)